staging/src/k8s.io/apimachinery/pkg/runtime/serializer/protobuf/protobuf.go

```diff
@@ -161,15 +162,25 @@ func (s *Serializer) Decode(originalData []byte, gvk *schema.GroupVersionKind, i
 	return unmarshalToObject(s.typer, s.creater, &actual, into, unk.Raw)
 }
 
+// EncodeWithAllocator writes an object to the provided writer.
+// In addition, it allows for providing a memory allocator for efficient memory usage during object serialization.
+func (s *Serializer) EncodeWithAllocator(obj runtime.Object, w io.Writer, memAlloc runtime.MemoryAllocator) error {
```

Comment:
I will apply the same technique to the RawSerializer.

Reply:
do you know why we can't switch the CRDs to proto?

Reply:
For protobufs, we rely on auto-generated per-type code that is compiled in. For CRDs, we clearly can't reuse the same approach.
[We believe it's possible to support it, but it's a larger effort on its own.]